apps/summon-app/src/utils/summonTx.ts

```diff
@@ -0,0 +1,265 @@
+  // without anything other than the shape seen below.
+  // React Hook Form only allows us to shape the functions with Record<string, unknown>
+
+  const memberData = formValues[FORM_KEYS.MEMBERS] as Record<string, unknown>;
+
+  if (
+    !isArray(memberData?.memberAddresses) ||
+    memberData.memberAddresses.some((addr) => !isString(addr)) ||
+    !isArray(memberData?.memberShares) ||
+    memberData.memberShares.some((shares) => !isNumberish(shares))
+  ) {
+    console.log('ERROR: Form Values', formValues);
+    throw new Error(
+      'shareConfigTX recieved arguments in the wrong shape or type'
+    );
+  }
+
+  const wholeShareAmts = memberData.memberShares as string[];
+  const sharesInBaseUnits = wholeShareAmts.map((shares) =>
+    toBaseUnits(shares.toString())
+  );
+  return encodeFunction(LOCAL_ABI.BAAL, 'mintShares', [
+    memberData.memberAddresses,
+    sharesInBaseUnits,
+  ]);
+};
+
+export const lootConfigTX = (formValues: Record<string, unknown>) => {
+  const memberData = (formValues[FORM_KEYS.MEMBERS] || {}) as Record<
+    string,
+    unknown
+  >;
+
+  if (
+    !isArray(memberData?.memberAddresses) ||
+    memberData.memberAddresses.some((addr) => !isString(addr)) ||
+    !isArray(memberData?.memberLoot) ||
+    memberData.memberLoot.some((loot) => !isNumberish(loot))
+  ) {
+    console.log('ERROR: Form Values', formValues);
+    throw new Error(
+      'shareConfigTX recieved arguments in the wrong shape or type'
+    );
+  }
+
+  const wholeLootAmts = memberData.memberLoot as (string | number)[];
+  const lootInBaseUnits = wholeLootAmts.map((loot) =>
+    toBaseUnits(loot.toString())
+  );
+  return encodeFunction(LOCAL_ABI.BAAL, 'mintLoot', [
+    memberData.memberAddresses,
+    lootInBaseUnits,
+  ]);
+};
+
+const metadataConfigTX = (
+  formValues: Record<string, unknown>,
+  posterAddress: string
+) => {
+  const daoName = formValues[FORM_KEYS.DAO_NAME];
+  if (!isString(daoName)) {
+    console.log('ERROR: Form Values', formValues);
+    throw new Error('metadataTX recieved arguments in the wrong shape or type');
+  }
+
+  const METADATA = encodeFunction(LOCAL_ABI.POSTER, 'post', [
+    JSON.stringify({ name: daoName }),
+    'daohaus.metadata.summoner',
+  ]);
+
+  return encodeFunction(LOCAL_ABI.BAAL, 'executeAsBaal', [
+    posterAddress,
+    0,
+    METADATA,
+  ]);
+};
+
+const handleKeychains = (chainId: ValidNetwork) => {
+  const {
+    V3_FACTORY,
+    V3_LOOT_SINGLETON,
+    V3_SHARE_SINGLETON,
+    GNOSIS_MULTISEND,
+    POSTER,
+  } = CONTRACTS;
+  const v3Contracts = [
+    V3_FACTORY,
+    V3_LOOT_SINGLETON,
+    V3_SHARE_SINGLETON,
+    GNOSIS_MULTISEND,
+    POSTER,
+  ];
+
+  if (v3Contracts.every((contract) => contract[chainId])) {
+    // REVIEW. Before this conditional statement, each contract
+    // was string | undefined. After the conditional statement it can
+    // only be string. So I'm typecasting here.
```

Comment:
There is probably some disconnect here with typescript. If we are looping through every item it might make sense to combine into a netowork object ad then just access the keys to avoid the typecasting